Program.cs

```diff
@@ -150,97 +130,44 @@ private static void StartProcess(string relativePath)
                         + Environment.NewLine + Environment.NewLine +
                         "If the client still doesn't run, please contact the mod's authors for support.",
                         "Client Launcher Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
-                    return;
                 }
             }
         }
 
         private static OSVersion GetOperatingSystemVersion()
         {
-            Version osVersion = Environment.OSVersion.Version;
-
-            if (Environment.OSVersion.Platform == PlatformID.Win32Windows)
-                return OSVersion.WIN9X;
-
             if (Environment.OSVersion.Platform == PlatformID.Win32NT)
-            {
-                if (osVersion.Major < 5)
-                    return OSVersion.UNKNOWN;
-
-                if (osVersion.Major == 5)
-                    return OSVersion.WINXP;
-
-                if (osVersion.Minor > 1)
-                    return OSVersion.WIN810;
-                else if (osVersion.Minor == 0)
-                    return OSVersion.WINVISTA;
-
-                return OSVersion.WIN7;
-            }
-
-            int p = (int)Environment.OSVersion.Platform;
-
-            if ((p == 4) || (p == 6) || (p == 128))
-            {
-                return OSVersion.UNIX;
-            }
+                return OSVersion.WIN;
 
             return OSVersion.UNKNOWN;
         }
 
-        private static bool IsNetFramework4Installed()
+        private static bool IsXNAFramework4RefreshInstalled()
         {
             try
             {
-                RegistryKey ndpKey = Registry.LocalMachine.OpenSubKey("SOFTWARE\\Microsoft\\NET Framework Setup\\NDP\\v4\\Full");
+                RegistryKey xnaKey = Registry.LocalMachine.OpenSubKey("SOFTWARE\\Microsoft\\XNA\\Framework\\v4.0");
 
-                string installValue = ndpKey.GetValue("Install").ToString();
+                string installValue = xnaKey?.GetValue("Refresh1Installed").ToString();
 
                 if (installValue == "1")
                     return true;
             }
             catch
             {
-                
             }
 
-            return false;
-        }
-
-        private static bool IsNetFramework45Installed()
-        {
             try
             {
-                RegistryKey ndpKey = Registry.LocalMachine.OpenSubKey("SOFTWARE\\Microsoft\\NET Framework Setup\\NDP\\v4\\Full");
+                RegistryKey xnaKey = Registry.LocalMachine.OpenSubKey("SOFTWARE\\WOW6432Node\\Microsoft\\XNA\\Framework\\v4.0");
```

Comment:
Don't do that. See #1 for the correct use

Reply:
In other words, `WOW6432Node` should never be a part of the registry key path. Let operating system redirect it by specifying options when opening a key.